numpy/lib/function_base.py

```diff
@@ -944,11 +944,15 @@ def piecewise(x, condlist, funclist, *args, **kw):
             condlist = condlist.T
     if n == n2 - 1:  # compute the "otherwise" condition.
         totlist = np.logical_or.reduce(condlist, axis=0)
-        condlist = np.vstack([condlist, ~totlist])
+        try:
```

Comment:
Yes it should be possible as we can check for the number of dimensions. Thanks for the feedback. I agree the `try...except` block is ugly and even more so without a specific exception.